File: day_30/password_generator.py
#the password generator project

#required modules
from random import choice, shuffle

#function for generating a character
def generate(lst,num):
    for _ in range(num):
        password.append(choice(char[lst]))

#character sets
letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z']
numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
symbols = ['!', '#', '$', '%', '&', '(', ')', '*', '+']
#stored in another list for referencing easily via index
char=[letters,numbers,symbols]       

#user greeting and input
# print("Welcome to the PyPassword Generator!")
# nr_letters = int(input("How many letters would you like in your password?\n")) 
# nr_symbols = int(input(f"How many symbols would you like?\n"))
# nr_numbers = int(input(f"How many numbers would you like?\n"))

#creating list for password characters
password=[]

#generating password characters in order
def generate_pwd():
    global password
    password = []
    generate(0,4)
    generate(1,4)
    generate(2,4)

    #shuffling the order of the list
    shuffle(password)

    #joining elements of the password list into a string
    pwd=""
    for ch in password:
        pwd+=ch

    
    #printing final password string
    # print(pwd)
    return pwd

File: day_5/password_generator.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <time.h>

using namespace std;

void append(char x,char arr[],int &z)
{
    arr[z]=x;
    z++;
}

int array_len(char array[])
{
    int i=0,count=0;
    while(i==0)
    {
        if(array[i]!='\0')
        {
            count++;
            i++;
        }
        else
            break;
    }
    return count;
}

int main()
{
    //variable declaration and assignment
    char letters[]= {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'};
    char numbers[] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9'};
    char symbols[] = {'!', '#', '$', '%', '&', '(', ')', '*', '+'};
    int nr_letters, nr_symbols, nr_numbers;
    int l_size=sizeof(letters);
    int n_size=sizeof(numbers);
    int s_size=sizeof(symbols);

    //user greeting and input 
    printf("Welcome to the PyPassword Generator!\n\n");
    printf("How many letters would you like in your password? ") ;
    cin>>nr_letters;
    printf("How many symbols would you like? ");
    cin>>nr_symbols;
    printf("How many numbers would you like? ");
    cin>>nr_numbers;
    
    //password array size determined by input size
    char password[nr_letters+nr_numbers+nr_symbols];
    int password_array_size=sizeof(password);
    int password_length;
    password_length=array_len(password);

    int i=0;
    for(i=0;i<nr_numbers;i++)
        cout<<numbers[rand() % sizeof(numbers)];
    for(i=0;i<nr_letters;i++)
        cout<<letters[rand() % sizeof(letters)];
    for(i=0;i<nr_symbols;i++)
        cout<<symbols[rand() % sizeof(symbols)];
    

}